### PR TITLE
fix(net, discv4): call find_node with valid endpoint

### PR DESCRIPTION
**Pull request by fgimenez**
Prevents calling `find_node()` for a node record that does not yet have a valid endpoint.

The case can be reached if the other user has sent a Ping but not sent a Pong response. They will be added to the DHT and are considered a valid entry in `self.kbuckets.entry()`.

The function `on_neighbours()` will then iterate through all entries including those with `has_endpoint_proof = false` and if this node is the closest it will reach out to that node via `find_node`.

To fix it, in `on_neighbours()` first check that the closest entry has `has_endpoint_proof = true` before calling `find_node()` for that endpoint.

### CODE REVIEW

File: crates/net/discv4/src/lib.rs [hunk 1]
```diff
@@ -1387,7 +1387,22 @@ impl Discv4Service {
                 BucketEntry::SelfEntry => {
                     // we received our own node entry
                 }
-                _ => self.find_node(&closest, ctx.clone()),
+                _ => {
+                    let key = kad_key(closest.id);
```
Comment (onbjerg) [inline crates/net/discv4/src/lib.rs:1391]: we already have `key` defined just before the match statement, and we also already do a lookup in `self.kbuckets` here, so we can make this two match arms instead

**Review by onbjerg — CHANGES_REQUESTED** [review]
see comment

File: crates/net/discv4/src/lib.rs (hunk 1 above)
Comment (fgimenez) [reply crates/net/discv4/src/lib.rs]: totally missed it thx! done here https://github.com/paradigmxyz/reth/pull/8002/commits/b2a0b997ec90e14d6d00440047e631c2008007c7

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by mattsse — APPROVED** [review]

**Pull request merged**